huaweicloud/services/coc/data_source_huaweicloud_coc_war_rooms.go

```diff
@@ -0,0 +1,508 @@
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"incident_id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"is_change_event": {
+					Type:     schema.TypeBool,
+					Computed: true,
+				},
+				"failure_level": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"incident_url": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+}
+
+func buildWarRoomsReqRegions() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"code": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+}
+
+func buildWarRoomsReqWarRoomStatus() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"name_zh": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"name_en": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"type": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+}
+
+func buildWarRoomsReqImpactedApplication() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+}
+
+// @API COC POST /v1/external/warrooms/list
+func dataSourceCocWarRoomsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	var mErr *multierror.Error
+
+	client, err := cfg.NewServiceClient("coc", cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating COC client: %s", err)
+	}
+
+	listHttpUrl := "v1/external/warrooms/list"
+	listPath := client.Endpoint + listHttpUrl
+
+	listOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	currentOffset := 0
+	list := make([]map[string]interface{}, 0)
+	for {
+		listOpt.JSONBody = utils.RemoveNil(buildListWarRoomsBodyParams(d, currentOffset))
+		listResp, err := client.Request("POST", listPath, &listOpt)
+		if err != nil {
+			return diag.Errorf("error retrieving COC war rooms: %s", err)
+		}
+		listRespBody, err := utils.FlattenResponse(listResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		listWarRooms, nextOffset := flattenCocListWarRooms(listRespBody, currentOffset)
```

Comment:
can use currentOffset+=len(listWarRooms) to get nextoffset
```suggestion
		listWarRooms := flattenCocListWarRooms(listRespBody)
```